### DISCUSSION TITLE
Deviation between the half cell model and full DFN

### DISCUSSION THREAD

**Discussion by thuwangming** [pybamm-team/PyBaMM#2067] [General]
Hello, 
I am trying to compare the half-cell model and the full DFN model, and I get some weird results. 
I use the same parameter set to build a half-cell model and a full model, respectively. And some parameters for lithium plating are manually added from parameter set "Xu2019". When I run the models, I find that the terminal voltage of the half cell always dropped quickly below the cut-off voltage, but the full model looks normal. I have tested two parameter sets: "Marquis2019" and "Chen2020". The results and source codes are attached below. 
How can I fix the problem? Thanks!

Full cell for "Chen2020"
![1](https://user-images.githubusercontent.com/1240636/169456276-01560ae0-5cf3-4c15-90d1-b76f8e8f4b4d.png)

Half cell for "Chen2020"
![2](https://user-images.githubusercontent.com/1240636/169456447-9905664f-2fbd-4afa-96d3-d320188f9960.png)

Full cell for "Marquis2019"
![4](https://user-images.githubusercontent.com/1240636/169456558-5d94e07e-199a-4c27-a416-ce38b6628b90.png)

Half cell for "Marquis2019"
![3](https://user-images.githubusercontent.com/1240636/169456664-60627f4e-4065-4b8f-b75d-40c07969f351.png)

`# coding: utf-8

import pybamm
import numpy as np
from pybamm.input.parameters.lithium_ion.negative_electrodes.li_metal_Xu2019.li_metal_electrolyte_exchange_current_density_Xu2019 import li_metal_electrolyte_exchange_current_density_Xu2019

pybamm.set_logging_level("INFO")

isHalfCell = False
parameter_set = "Marquis2019"
C_rate = 1

if isHalfCell:
    #Half Cell
    model = pybamm.lithium_ion.DFN({"working electrode": "positive"})
else:
    #Full DFN
    model = pybamm.lithium_ion.DFN()

geometry = model.default_geometry

param = pybamm.ParameterValues(parameter_set)
param.export_csv(parameter_set+".csv")
param.update({"Exchange-current density for plating [A.m-2]":li_metal_electrolyte_exchange_current_density_Xu2019}, check_already_exists=False)
param.update({"Typical plated lithium concentration [mol.m-3]":76900}, check_already_exists=False)
param['Current function [A]'] = C_rate * param['Current function [A]']

#param.process_geometry(geometry)
param.process_model(model)

# solve model
solver = pybamm.CasadiSolver(mode="safe", rtol=1e-6, atol=1e-6, dt_max=0.005, extra_options_setup={"max_num_steps": 50000})
t_eval = np.linspace(0, 3600, 100)
sim = pybamm.Simulation(model, solver=solver)
sim.solve(t_eval)

# plot
pybamm.dynamic_plot(sim, output_variables=[
    "Positive particle concentration [mol.m-3]",
    "Electrolyte concentration [mol.m-3]",
    "Interfacial current density [A.m-2]",
    "Electrolyte potential [V]",
    "Positive electrode potential [V]",
    "Terminal voltage [V]",
    "X-averaged positive electrode interfacial current density [A.m-2]",
    "Sum of x-averaged positive electrode interfacial current densities",
    "Positive particle surface concentration [mol.m-3]"
])
`

**Comment by TomTranter** [comment]
Something looks wrong with the thickness of the layer which is too small for half cell and interfacial current is therefore higher and capacity lower than expected

**Reply by thuwangming** [reply]
I have exported all the parameters, it looks the same.

**Reply by TomTranter** [reply]
The x-axis on the half cells looks to go from about 212 um to 225 um for the cathode whereas it's more like 100 um thick in the full cell. There may be some bug in the half cell code scaling this length badly... @brosaplanella would you know where to look?

**Reply by thuwangming** [reply]
@TomTranter Yes, I noticed that. So I try to get the actual parameter values used in the simulation from the model, but it seems the model class does not implement the method.

**Comment by valentinsulzer** [comment]
You are not passing the parameter values to the simulation, so in the first instance it uses the default for a full-cell model (Marquis2019), and in the second it uses the default for a half-cell model (Xu2019) for the geometry. Try with `simulation = pybamm.Simulation(model, solver=solver, parameter_values=param)` instead of processing the model first.

**Reply by thuwangming** [reply]
@tinosulzer Thanks, that works for me. But when should I use the ParameterValues.process_model() method?
